src/ts/actions.ts

```diff
@@ -81,6 +86,14 @@ export function requestFromFunction(
   };
 }
 
+export function requestWithConfig(
+  actionSet: AsyncActionSet,
+  axoisOptions: AxiosRequestConfig,
+  params: ExtendedRequestParams = {}
```

Comment:
These params are not like the existing params. They are the meta data, so should not contain a `metaData` key.

Some refactoring should be made inside `requestFromFunction` so that it doesn't extract and combine different pieces of meta.

Reply:
As far as I can see, the metaData is not the metaData that is being sent to the Axios function, but rather some extra metadata that is sent with the action, I could remove that but I guess that would remove some extra functionality from the library.

Reply:
From the readme:
```
...
metaData - for storing additional data about the request that will not be forwarded to the server.
...
```

Reply:
I don't really understand the request, do you want me to refactor out the metadata out from the requestparams? it does still include the tag and shouldRethrough attribute that will be used in the requestFromFunction function, so those will still need to be sent in somehow.